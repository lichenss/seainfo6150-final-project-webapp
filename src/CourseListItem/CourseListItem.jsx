import React from 'react';
import styles from './CourseListItem.module.css';
import { Link } from 'react-router-dom';

const CourseListItem = (props) => {
    var data={
        title:props.course.title,
        imgurl:props.course.image._url,
        description:props.course.description,
        instructor:props.course.instructor
    };
    

    return (
        <li className={styles.course}>
            <Link
                className={styles.link}
                to={{
                    pathname: `/category/${props.category}/course/${props.course.slug}`,
                    state:data,
                }}
            >
                <div className={styles.imageBlock}>
                    <img
                        className={styles.image}
                        src={props.course.image._url}
                        alt={`${props.course.title}`}
                    />
                </div>
                <div className={styles.courseInfoBlock}>
                    <h3 className={styles.title}>{props.course.title}</h3>
                </div>
            </Link>
        </li>
    );
};
export default CourseListItem