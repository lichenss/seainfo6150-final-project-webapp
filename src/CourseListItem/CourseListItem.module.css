.course {
  width: 80%;
}


.imageBlock {
  width: 100%;
  height: 8rem;
}

.image {
  width: 100%;
  height: 100%;
  object-fit: cover;
}

.title {
  font-family: 'Open Sans', sans-serif;
  font-size: 1.5rem;
  margin-top: 8px;
  color: black;
}

@media screen and (min-width:600px){
  .imageBlock {
    height: 4rem;
  }
}
@media screen and (min-width:900px){
  .imageBlock {
    height: 6rem;
  }
}
@media screen and (min-width:1200px){
  .imageBlock {
    height: 8rem;
  }
}
