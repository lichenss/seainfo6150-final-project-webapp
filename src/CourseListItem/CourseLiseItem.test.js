import React from "react";
import CourseListItem from "./CourseListItem";
import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
const course={
    "slug": "art-0001",
"title": "Therapeutic Art Life Coach Certification (Accredited)",
"description": "Therapeutic Art: Help Clients Heal, Reduce Stress & Manifest Dreams with Mandalas, Affirmations, Vision Boards & More",
"instructor": "Janet Giessl",
"image": {
  "_url": "https://arhu.umd.edu/sites/default/files/2019-04/iStock-955549248%20composite-JC%20flat2%20opt1-min.jpg",
  "_height": "619",
  "_width": "1100"
},
"requirements":"Ability to print course material."
}
describe("CourseListItem tests", () => {
    it("renders correctly", () => {
      const { container } = render(
      <BrowserRouter>
      <CourseListItem course={course} category='computer-science' />
      </BrowserRouter>);
      expect(container).toMatchSnapshot();
    });
  });