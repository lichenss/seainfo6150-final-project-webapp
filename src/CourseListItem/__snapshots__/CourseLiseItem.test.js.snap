// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseListItem tests renders correctly 1`] = `
<div>
  <li
    class="course"
  >
    <a
      class="link"
      href="/category/computer-science/course/art-0001"
    >
      <div
        class="imageBlock"
      >
        <img
          alt="Therapeutic Art Life Coach Certification (Accredited)"
          class="image"
          src="https://arhu.umd.edu/sites/default/files/2019-04/iStock-955549248%20composite-JC%20flat2%20opt1-min.jpg"
        />
      </div>
      <div
        class="courseInfoBlock"
      >
        <h3
          class="title"
        >
          Therapeutic Art Life Coach Certification (Accredited)
        </h3>
      </div>
    </a>
  </li>
</div>
`;
